### PR TITLE
fix: add bfloat16 to allowed dtypes

### PR DESCRIPTION
**Pull request by samedii**
This was very painful to debug. Maybe it shouldn't have been but this was unexpected for me.

Sorry about all the formatting fixes. It's black. The change is just adding "torch.bfloat16" to the device types. Also bumped the version

### CODE REVIEW

File: ema_pytorch/ema_pytorch.py [hunk 1]
```diff
@@ -86,21 +92,35 @@ def __init__(
...
+                    "Your model was not copyable. Please make sure you are not using any LazyLinear"
+                )
                 exit()
 
         self.ema_model.requires_grad_(False)
 
         # parameter and buffer names
 
-        self.parameter_names = {name for name, param in self.ema_model.named_parameters() if param.dtype in [torch.float, torch.float16]}
-        self.buffer_names = {name for name, buffer in self.ema_model.named_buffers() if buffer.dtype in [torch.float, torch.float16]}
```
Comment (lucidrains) [inline ema_pytorch/ema_pytorch.py:98]: ah, maybe should just use `torch.is_floating_point` here. to cover complex, maybe an `or torch.is_complex` as well

Comment (lucidrains) [reply ema_pytorch/ema_pytorch.py:98]: nice, if you redo the PR with just these 2 loc change as well as version bump, will accept

Comment (samedii) [reply ema_pytorch/ema_pytorch.py:98]: I don't mind if you do the modifications

**Comment by w-hc** [comment]
Hi a quick question: is bfloat16's precision enough for common EMA coefficients?

**Comment by samedii** [comment]
bfloat16 has the same precision as float32 (but its precision on integers is bad)

**Comment by w-hc** [comment]
> has the same precision as float32

I thought they share the same range (both 8 bits). Precision is much worse (23 bits vs 7 bits).

**Comment by samedii** [comment]
Sorry you're right. I thought the integer was 16 bit

**Pull request closed without merging**